# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../xdm/startwm.patch
# Copyright (C) 2010 The OpenSDE Project
# Copyright (C) 2004 - 2006 The T2 SDE Project
# Copyright (C) 1998 - 2004 Clifford Wolf
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

More flexible root application startup.

diff -ruN xdm-1.1.9-orig/config/Xsetup_0 xdm-1.1.9/config/Xsetup_0
--- xdm-1.1.9-orig/config/Xsetup_0	2009-09-18 04:24:37.000000000 +0200
+++ xdm-1.1.9/config/Xsetup_0	2010-07-23 15:37:43.568753394 +0200
@@ -1,2 +1,3 @@
 #!/bin/sh
 xconsole -geometry 480x130-0-0 -daemon -notify -verbose -fn fixed -exitOnFail
+xsetroot -solid black
diff -ruN xdm-1.1.9-orig/config/Xsession.cpp xdm-1.1.9/config/Xsession.cpp
--- xdm-1.1.9-orig/config/Xsession.cpp	2009-09-18 04:24:37.000000000 +0200
+++ xdm-1.1.9/config/Xsession.cpp	2010-07-23 15:37:43.568753394 +0200
@@ -62,23 +62,13 @@
 	if [ -r "$resources" ]; then
 		BINDIR/xrdb -load "$resources"
 	fi
-#if defined(__SCO__) || defined(__UNIXWARE__)
-        [ -r /etc/default/xdesktops ] && {
-                . /etc/default/xdesktops
-        }
 
-        [ -r /etc/default/xdm ] && {
-                . /etc/default/xdm
-        }
+	[ -f /etc/profile ] && . /etc/profile
 
-        XCOMM Allow the user to over-ride the system default desktop
-        [ -r $HOME/.xdmdesktop ] && {
-                . $HOME/.xdmdesktop
-        }
+	[ "$WINDOWMANAGER" =  "gnome" ] && WINDOWMANAGER="gnome-session"
+	[ "$WINDOWMANAGER" =  "kde"   ] && WINDOWMANAGER="startkde"
+	[ "$WINDOWMANAGER" != "twm"   ] && exec $WINDOWMANAGER
+	xsetroot -solid darkcyan ; exec twm ; exec xterm
 
-        [ -n "$XDESKTOP" ] && {
-                exec `eval $XDESKTOP`
-        }
-#endif
 	exec BINDIR/xsm
 fi
